### PR TITLE
ENH: add support for direction in VTK image

### PR DESCRIPTION
**Pull request by dave3d**
I have added support for getting the VTK direction matrix when converting
it to an ITK image in Python and for setting it when going in the other direction.

## PR Checklist
- [ ] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [ ] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Review by dzenanz — COMMENTED** [review]
LGTM, but someone else should review too.

File: Wrapping/Generators/Python/itk/support/extras.py [hunk 1]
```diff
@@ -773,6 +783,21 @@ def image_from_vtk_image(vtk_image: "vtk.vtkImageData") -> "itkt.ImageBase":
     l_origin[:dim] = vtk_image.GetOrigin()[:dim]
     l_image.SetOrigin(l_origin)
     # Todo: Add Direction with VTK 9
+    import vtk
+    if vtk.vtkVersion.GetVTKMajorVersion()>=9:
+        direction = vtk_image.GetDirectionMatrix()
+        if dim==3:
+            direction_array = np.identity(3)
+            for y in [0,1,2]:
```
Comment (thewtex) [inline Wrapping/Generators/Python/itk/support/extras.py:791]: As a minor performance tweak, we can use a tuple over a list here and below.

File: Wrapping/Generators/Python/itk/support/extras.py [hunk 2]
```diff
@@ -773,6 +783,21 @@ def image_from_vtk_image(vtk_image: "vtk.vtkImageData") -> "itkt.ImageBase":
...
+                for x in [0,1,2]:
+                    direction_array[x,y] = direction.GetElement(x,y)
+        if dim==2:
```
Comment (thewtex) [inline Wrapping/Generators/Python/itk/support/extras.py:794]: `if` -> `elif`

**Review by thewtex — CHANGES_REQUESTED** [review]
@dave3d thanks for the contribution!

Can a test with a non-trivial direction please be added to *ITK/Wrapping/Generators/Python/Tests/extras.py*?

**Comment by dave3d** [comment]
Yeah, I can do some sort of test.

**Review by thewtex — APPROVED** [review]
@dave3d looks great! 🥇 

Could the branch please be rebased on `master`, which will hopefully address CI tests?

**Comment by dzenanz** [comment]
It looks like the rebase didn't go smoothly, it pulled in 40+ commits.

**Comment by dave3d** [comment]
> It looks like the rebase didn't go smoothly, it pulled in 40+ commits.

Yeah, should I just blow away this PR and make a new one?  I only touched 2 files.  As @blowekamp can tell you, I'm not an expert git user.

**Comment by dzenanz** [comment]
If you use some visual interface for git, such as `gitk` or `TortoiseGit`, it is easy to see what you are doing. Botching git operations when only using command line is easy, at least for me.

**Comment by lassoan** [comment]
No need to close the PR (it would just add noise), just fetch & rebase on latest ITK master and force-push to the same branch. I, too, use TortoiseGit as git GUI and it makes these operations very easy to do.

File: Wrapping/Generators/Python/itk/support/extras.py [hunk 3]
```diff
@@ -782,19 +782,33 @@ def image_from_vtk_image(vtk_image: "vtk.vtkImageData") -> "itkt.ImageBase":
     l_origin = [0.0] * dim
     l_origin[:dim] = vtk_image.GetOrigin()[:dim]
     l_image.SetOrigin(l_origin)
+<<<<<<< HEAD
```
Comment (dzenanz) [inline Wrapping/Generators/Python/itk/support/extras.py:785]: Unresolved conflicts?

Comment (dave3d) [reply Wrapping/Generators/Python/itk/support/extras.py]: yeah, clearly I've botched it up.  trying to fix it. :)

**Comment by thewtex** [comment]
@dave3d good job getting this updated. #2614 should address CI issues, so rebasing skills can be re-applied :-)

**Comment by thewtex** [comment]
@dave3d can this please be rebased on `master`?

**Review by thewtex — APPROVED** [review]
:star2:

**Pull request merged**